core/governance/engine.go

```diff
@@ -794,7 +795,7 @@ func (e *Engine) validateOpenProposal(proposal *types.Proposal) (types.ProposalE
 
 	checkProposerToken := true
 
-	if proposal.IsMarketUpdate() {
+	if proposal.IsMarketUpdate() || proposal.IsMarketStateUpdate() {
 		proposalError, err := e.validateMarketUpdate(proposal, params)
```

Comment:
This is going to panic. When we call `validateMarketUpdate()` if its an MarketStateUpdate proposal.